File: 08_04_2024_Number_of_Students_Unable_to_Eat_Lunch.py
class Solution:
    def countStudents(self, students, sandwiches):
        n = len(students)
        ans = 0
        stu = defaultdict(int)
        sand = defaultdict(int)

        for i in range(n):
            stu[students[i]] += 1

        for i in range(n):
            sand[sandwiches[i]] += 1
            if sand[sandwiches[i]] > stu[sandwiches[i]]:
                return n - i

        return ans

File: 08_04_2024_Number_of_Students_Unable_to_Eat_Lunch.cpp
class Solution {
public:
    int countStudents(vector<int>& students, vector<int>& sandwiches) {
        int n = students.size(), ans = 0;
        unordered_map<int,int> stu , sand;
        for(int i = 0 ; i  <n ; i++)
        {
            stu[students[i]]++;
        }
        for(int i = 0 ; i < n ; i++)
        {
            sand[sandwiches[i]]++;
            if(sand[sandwiches[i]]>stu[sandwiches[i]])
            {
                return n-i;
            }
        }
        return ans;
    }
};
